doc/platform-optimization/README.md

```diff
@@ -0,0 +1,188 @@
+for containers and sets of containers. This ensures that critical
+containers will always have enough compute resources, and that
+resource hungry containers will not be able to hurt all processes in
+the system.
+
+## NRI Plugins
+
+[NRI plugins](https://github.com/containers/nri-plugins) connect to
+the container runtime running on a Kubernetes node. Containerd and
+CRI-O runtimes support NRI plugins.
+
+The NRI plugins project includes two resource policies, balloons and
+topology-aware. They manage allowed CPUs and memories (cpuset.cpus and
+cpuset.mems) of all Kubernetes containers created and running on the
+node.
+
+In this example, we use the balloons policy because it can be tuned
+for certain applications (like RAG pipelines) using even node-specific
+parameters for each container in applications. The topology-aware
+policy, on the other hand, needs no configuration and does CPU
+assignment automatically based on resource requests in containers and
+underlying hardware topology.
+
+## Install
+
+Warning: installing and reconfiguring the balloons policy in can
+change allowed CPUs and memories of already running containers in the
+cluster. This may hurt containers that rely on the number of allowed
+CPUs being static.
+
+Install the balloons policy with helm:
+
+1. Add the NRI plugins repository
+   ```bash
+   helm repo add nri-plugins https://containers.github.io/nri-plugins
+   ```
+
+2. Install the balloons resource policy and patch container runtime's
+   configuration on host in order to switch NRI support on.
+   ```bash
+   helm install balloons nri-plugins/nri-resource-policy-balloons --set patchRuntimeConfig=true
+   ```
+
+Now the balloons policy is managing node resources in the cluster as a
+DaemonSet that communicates with the container runtime on every
+node. You should see `nri-resource-policy-balloons-...` pod running on
+every node.
+
+## Configure
+
+Edit the default balloons policy:
+```bash
+kubectl edit balloonspolicy default
+```
+
+Let us consider isolating AI inference and reranking containers in
+[ChatQnA](https://github.com/opea-project/GenAIExamples/tree/main/ChatQnA)
+application's Gaudi accelerated pipeline.
+
+In the
+[manifest](https://github.com/opea-project/GenAIExamples/blob/main/ChatQnA/kubernetes/manifests/gaudi/chatqna.yaml)
+there are "tgi" and "tei" containers that will need a lot of CPUs.
+
+We will allocate 16 CPUs (8 full CPU cores with two hyperthreads on
+each core) for each tgi container, and 32 CPUs (that is 16
+hyperthreaded cores) for each tei container. This is done with the
+following balloons policy configuration that is explained below.
+
+```yaml
+apiVersion: config.nri/v1alpha1
+kind: BalloonsPolicy
+metadata:
+  name: default
+spec:
+  allocatorTopologyBalancing: true
+  balloonTypes:
+  - name: tgi
+    allocatorPriority: high
+    minCPUs: 16
+    minBalloons: 1
+    preferNewBalloons: true
+    hideHyperthreads: true
```

Comment:
Line 118 says hidehyperthreads false and here true. May need to add a sentence or two on why this different balloon policy setting for tgi versus tei based on the application characteristics/behavior.

Reply:
Parameters of balloon type definitions written in a bit different order (always starting with name) to make it easier to notice that hideHyperthreads: false is defined for the default balloon only.

I also added explanation of hideHyperthreads: false separately. There it was also worth mentioning that when hyperthreading is off in BIOS, hideHyperthreads: false does not unhide them.